// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

syntax = "proto3";

package qrl;

service P2PNode {
  rpc Ping (PingReq) returns (PongResp) {}
  rpc GetKnownPeers (GetKnownPeersReq) returns (GetKnownPeersResp) {}
}

service PublicAPI {
  rpc GetKnownPeers (GetKnownPeersReq) returns (GetKnownPeersResp) {}
  rpc GetAddressState (GetAddressStateReq) returns (GetAddressStateResp) {}
}

// API Messages

message PingReq {   string name = 1;    }
message PongResp {   string message = 1; }
message GetKnownPeersReq {}
message GetKnownPeersResp { KnownPeers known_peers = 1; }

message GetAddressStateReq {   string name = 1;    }
message GetAddressStateResp {   string message = 1; }

// Content

message WalletStore
{
    repeated Wallet wallets = 1;
}

message Wallet
{
    string address = 1;
    string mnemonic = 2;
    int32 xmss_index = 3;
}

message KnownPeers
{
    repeated Peer peers = 1;
}

message Peer
{
    string ip = 1;
}

